### PR TITLE
CROSSDATA-873 nativeudfs

### PR DESCRIPTION
**Pull request by pfcoperez**
This PR enables the push down of native built-in functions in C* while updating crossdata-core with the plumbing needed to expand the supported functions set or implement native functions push-downs for other data sources.

*NOTE*
Pushed-down native functions can be used:
- Within filters comparisons. e.g: `SELECT * FROM table01 WHERE colA = FUNCTION1(colB,1,"str")`
- As selection values within projections: `SELECT FUNCTION2(colB), colA FROM table01`.

### CODE REVIEW

File: cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala [hunk 1]
```diff
@@ -32,28 +37,46 @@ object CassandraQueryProcessor {
...
     def quoteString(in: Any): String = in match {
       case s: String => s"'$s'"
+      case a: Attribute => expandAttribute(a.toString)
       case other => other.toString
     }
 
+    def expandAttribute(att: String): String = {
+      udfs get(att) map { udf =>
+        val actualParams = udf.children.collect { //TODO: Add type checker (maybe not here)
+          case at: NativeUDFAttribute => expandAttribute(at.toString)
```
Comment (darroyocazorla) [inline cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala]: Are aliases supported?

File: cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala [hunk 2]
```diff
@@ -67,12 +90,18 @@ class CassandraQueryProcessor(cassandraRelation: CassandraXDSourceRelation, logi
...
+          columnsRequired.map(_.toString),
+          filters,
+          limit.getOrElse(CassandraQueryProcessor.DefaultLimit),
+          udfs map { case (k,v) => k.toString -> v}
+        )
         val resultSet = cassandraRelation.connector.withSessionDo { session =>
           session.execute(cqlQuery)
         }
-        sparkResultFromCassandra(columnsRequired, resultSet)
+        sparkResultFromCassandra(columnsRequired.map(_.name), resultSet)
```
Comment (darroyocazorla) [inline cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala]: Can we retrieve multiple duplicated UDFs? I mean, execute a SELECT now(), now() FROM _;

File: cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/DefaultSource.scala [hunk 3]
```diff
@@ -166,6 +157,19 @@ class DefaultSource extends CassandraConnectorDS with TableInventory {
...
   }
 
+  override def nativeBuiltinFunctions: Seq[UDF] = {
+
+    //TODO: Complete the built-in function inventory
+    Seq(
+      UDF("now", None, StructType(Nil), DataTypes.StringType),
+      UDF("dateOf", None, StructType(StructField("date",DataTypes.StringType, false)::Nil), DataTypes.StringType),
+      UDF("toDate", None, StructType(StructField("date",DataTypes.StringType, false)::Nil), DataTypes.StringType)/*,
+      UDF("", None, StructType(Array.empty), DataTypes.StringType),
```
Comment (darroyocazorla) [inline cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/DefaultSource.scala]: Remove comments

**Comment by jjlopezm** [comment]
+1

File: cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala (hunk 2 above)
Comment (pfcoperez) [inline cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala]: @darroyocazorla Yes

File: cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala (hunk 1 above)
Comment (pfcoperez) [inline cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala]: @darroyocazorla Yes but they won't be reflected at `CassandraSQLRow` level

**Comment by miguel0afd** [comment]
LGTM
+1

**Pull request merged**